src/rc_string.c

```diff
@@ -110,9 +123,68 @@ IMPLEMENT_MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create, const char*,
 
 THANDLE(RC_STRING) rc_string_create_with_format(const char* format, ...)
 {
-    (void)format;
+    THANDLE(RC_STRING) result = NULL;
 
-    return NULL;
+    if (format == NULL)
+    {
+        /*Codes_SRS_RC_STRING_07_001: [ If format is NULL, rc_string_create_with_format shall fail and return NULL. ]*/
+        LogError("Invalid arguments: const char* format=%s", MU_P_OR_NULL(format));
+    }
+    else
+    {
+        /*Codes_SRS_RC_STRING_07_002: [ Otherwise, `rc_string_create_with_format` shall determine the total number of characters written using the variable number of arguments. ]*/
+        va_list args;
+        va_start(args, format);
+        int string_length_with_terminator = vsnprintf(NULL, 0, format, args) + 1;
+        
+        if (string_length_with_terminator <= 0)
+        {
+            /*Codes_SRS_RC_STRING_07_003: [ If `vsnprintf` failed to determine the total number of characters written, `rc_string_create_with_format` shall fail and return `NULL`. ]*/
+            LogError("vsnprintf failed to determine the total number of characters written.");
+        }
+        else
+        {   
+            size_t memory_size_need = sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator;
+            
+            if (memory_size_need > SIZE_MAX) 
+            {
+                /*Codes_SRS_RC_STRING_07_009: [ If the resulting memory size requested for the `THANDLE(RC_STRING)`and the resulting formatted string results in an size_t overflow in `malloc_flex`, `rc_string_create_with_format` shall failand return `NULL`. ]*/
+                LogError("Requested memory size overflow.");
```

Comment:
>or("Requested memory size overflow."); [](http://example.com/codeflow?start=22&length=38)

Let's print the string_length_with_terminator for example so we know how much memory was requested.